docs/production.md

```diff
@@ -0,0 +1,420 @@
+    socket_keepalive_options={
+        1: 1,  # TCP_KEEPIDLE
+        2: 3,  # TCP_KEEPINTVL
+        3: 5,  # TCP_KEEPCNT
+    }
+)
+
+async with Docket(name="orders", connection_pool=pool) as docket:
+    # Use the custom pool
+    pass
+```
+
+### Redis Cluster and High Availability
+
+Docket works with Redis clusters and sentinel configurations:
+
+```python
+# Redis Cluster
+docket_url = "redis://redis-cluster:6379/0?cluster=true"
+
+# Redis Sentinel
+docket_url = "redis+sentinel://sentinel1:26379,sentinel2:26379/mymaster"
+
+# With authentication
+docket_url = "redis://:password@redis.prod.com:6379/0"
+```
+
+### Valkey Support
+
+Docket also works with Valkey (Redis fork):
+
+```bash
+export DOCKET_URL=valkey://valkey.prod.com:6379/0
+```
+
+## Monitoring and Observability
+
+### Prometheus Metrics
+
+Enable Prometheus metrics with the `--metrics-port` option:
+
+```bash
+docket worker --metrics-port 9090
+```
+
+Available metrics include:
+
+**Task Counters:**
+- `docket_tasks_added` - Tasks scheduled
+- `docket_tasks_started` - Tasks begun execution
+- `docket_tasks_succeeded` - Successfully completed tasks
+- `docket_tasks_failed` - Failed tasks
+- `docket_tasks_retried` - Retry attempts
+- `docket_tasks_stricken` - Tasks blocked by strikes
+
+**Task Timing:**
+- `docket_task_duration` - Histogram of task execution times
+- `docket_task_punctuality` - How close tasks run to their scheduled time
+
+**System Health:**
+- `docket_queue_depth` - Tasks ready for immediate execution
+- `docket_schedule_depth` - Tasks scheduled for future execution
+- `docket_tasks_running` - Currently executing tasks
+- `docket_redis_disruptions` - Redis connection failures
+- `docket_strikes_in_effect` - Active strike rules
+
+All metrics include labels for docket name, worker name, and task function name.
+
+### Health Checks
+
+Enable health check endpoints:
+
+```bash
+docket worker --healthcheck-port 8080
+```
+
+The health check endpoint (`/`) returns 200 OK when the worker is healthy and able to process tasks.
+
+### OpenTelemetry Traces
+
+Docket automatically creates OpenTelemetry spans for task execution:
+
+- **Span name**: `docket.task.{function_name}`
+- **Attributes**: docket name, worker name, task key, attempt number
+- **Status**: Success/failure with error details
+- **Duration**: Complete task execution time
+
+Configure your OpenTelemetry exporter to send traces to your observability platform:
+
+```python
+from opentelemetry import trace
+from opentelemetry.exporter.jaeger.thrift import JaegerExporter
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+# Configure tracing before creating workers
+trace.set_tracer_provider(TracerProvider())
+jaeger_exporter = JaegerExporter(
+    agent_host_name="jaeger",
+    agent_port=6831,
+)
+span_processor = BatchSpanProcessor(jaeger_exporter)
+trace.get_tracer_provider().add_span_processor(span_processor)
+```
```

Comment:
Rather than showing them in our docs, let's just link them out to https://opentelemetry.io/docs/languages/python/ and such